pkg/trace/agent/agent.go

```diff
@@ -296,11 +296,22 @@ func (a *Agent) ProcessStats(in pb.ClientStatsPayload, lang, tracerVersion strin
 	in.TracerVersion = tracerVersion
 	in.Lang = lang
 	for _, group := range in.Stats {
+		var blocked bool
+		i := 0
 		for _, b := range group.Stats {
 			normalizeStatsGroup(&b, lang)
+			if !a.Blacklister.AllowsStat(&b) {
+				blocked = true
```

Comment:
Please explain what's going on here.

Reply:
I added a comment. Please let me know if it's not clear or if you think this optimization adds too much complexity.

Reply:
You're right. The blocked thing was only avoiding cheap allocations and was adding complexity. I Removed it.